app/uploaders/banner_image_uploader.rb

```diff
@@ -0,0 +1,8 @@
+class BannerImageUploader < CarrierWave::Uploader::Base
+  include CarrierWave::Compatibility::Paperclip
+
+  # Define valid extensions for site banner image
+  def extension_white_list
+    %w(jpg jpeg png gif tif tiff)
```

Comment:
5000px wide seems excessive to me (maybe not at some point in the future but for the near future, at least). No ordinary user is going to have a browser near that wide. And all other things being equal, an image with larger dimensions will be larger in filesize.

Resizing to something like 2200px seems reasonable to me. It would cover the vast majority of cases and in the rare case where a user has a wider viewport, I think the current relevant CSS will just scale the image up (not good for images in general but this will be a very rare case, I think).

Rather than requiring a minimum size, we could just have some help text associated with the file upload that says "Upload an image at least 2200 pixels wide for best results." If the user uploads something smaller and it doesn't look good because it's upscaled, that's their problem, we warned them.